src/mlpack/methods/gmm/eigenvalue_ratio_constraint.hpp

```diff
@@ -64,7 +64,12 @@ class EigenvalueRatioConstraint
     // Eigendecompose the matrix.
     arma::vec eigenvalues;
     arma::mat eigenvectors;
-    arma::eig_sym(eigenvalues, eigenvectors, covariance);
+    covariance = arma::symmatu(covariance);
+    if (!arma::eig_sym(eigenvalues, eigenvectors, covariance))
+    {
+      Log::Warn << "applying to constraint could not be accomplished."
```

Comment:
> If arma::eig_sym() fails, eigval is reset so that it has no elements -- see the eig_sym() documentation.

So I don't think we can recover from that, I guess instead we could throw a std::runtime_error?

Reply:
Yeah, I would agree with that.  The same is true for all the other situations too, in my opinion.  You could use `Log::Fatal` too, that throws a `std::runtime_error`; or you can directly throw a `std::runtime_error`---personally I don't have a preference which.

Reply:
The issue have been resolved, changed all occurrences of Log::warn with Log::fatal